File: app/src/main/java/com/miaadrajabi/menuxmlparser/java/data/source/AssetsXmlDataSource.java
package com.miaadrajabi.menuxmlparser.java.data.source;

import android.content.Context;
import com.miaadrajabi.menuxmlparser.java.data.xml.XmlModels;
import com.miaadrajabi.menuxmlparser.java.data.xml.XmlConfig;

import java.io.IOException;
import java.io.InputStream;

public class AssetsXmlDataSource {
    private final Context context;

    public AssetsXmlDataSource(Context context) {
        this.context = context;
    }

    public XmlModels read(String filename) throws IOException {
        if (filename == null) filename = DEFAULT_FILE;
        try (InputStream input = context.getAssets().open(filename)) {
            return XmlConfig.getXmlMapper().readValue(input, XmlModels.class);
        }
    }

    public XmlModels read() throws IOException {
        return read(DEFAULT_FILE);
    }

    public static final String DEFAULT_FILE = "dynamic_menus.xml";
}

File: app/src/main/java/com/miaadrajabi/menuxmlparser/data/source/AssetsXmlDataSource.kt
package com.miaadrajabi.menuxmlparser.data.source

import android.content.Context
import com.miaadrajabi.menuxmlparser.data.xml.DynamicMenusXml
import com.miaadrajabi.menuxmlparser.data.xml.XmlConfig

class AssetsXmlDataSource(private val context: Context) {
    fun read(filename: String = DEFAULT_FILE): DynamicMenusXml {
        context.assets.open(filename).use { input ->
            return XmlConfig.xmlMapper.readValue(input, DynamicMenusXml::class.java)
        }
    }

    companion object {
        const val DEFAULT_FILE = "dynamic_menus.xml"
    }
}


